provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/contract/mqtt/ContractAPIMqtt.java

```diff
@@ -127,19 +127,29 @@ public synchronized void open(RequestData requestData) throws ProvisioningDevice
             throw new ProvisioningDeviceConnectionException(new IllegalArgumentException("sslContext cannot be null"));
         }
 
-        try
+        if (requestData.isX509() || (requestData.getSasToken() != null && !requestData.getSasToken().isEmpty()))
         {
-            String username = String.format(MQTT_USERNAME_FMT, this.idScope, registrationId, SDKUtils.getServiceApiVersion(), SDKUtils.getServiceApiVersion());
+            try
+            {
+                String username = String.format(MQTT_USERNAME_FMT, this.idScope, registrationId, SDKUtils.getServiceApiVersion(), SDKUtils.getServiceApiVersion());
+                String password = null;
 
-            this.mqttConnection = new MqttConnection(this.hostname, registrationId, username, null, sslContext, this, this.useWebSockets);
-            this.mqttConnection.connect();
+                if (requestData.getSasToken() != null && !requestData.getSasToken().isEmpty())
```

Comment:
Anyway, the password will be passed into `MqttConnection` if `null`. The check here is meaningless.

Reply:
Good catch. I'll just pass in the requestData.getSasToken and the lower layer will handle the null check